.github/workflows/java_thin_client_compatibility.yaml

```diff
@@ -0,0 +1,117 @@
+name: Test Java Thin client against the released IMDG servers
+
+on:
+  workflow_dispatch:
+    inputs:
+      organization_name:
+        description: Default is hazelcast, but if you would like to run the workflow with your forked repo, set your github username
+        required: true
+        default: hazelcast
+      branch_name:
+        description: Name of the branch to test client from
+        required: true
+        default: master
+
+jobs:
+  setup_server_matrix:
+    name: Setup the server test matrix
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - name: Setup Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.9
+
+      - name: Checkout to scripts
+        uses: actions/checkout@v2
+
+      - name: Set server matrix
+        id: set-matrix
+        run: |
+          versions=$( python get_server_matrix.py )
+          echo "Versions from Python script: $versions"
+          filtered_versions=$(echo "$versions" | jq -c '[.[] | select(. > "5.2")]')
+          echo "Filtered versions: $filtered_versions"
+          echo "matrix=$filtered_versions" >> "$GITHUB_OUTPUT"
+
+  test_client:
+    needs: [setup_server_matrix]
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        server-version: ${{ fromJson(needs.setup_server_matrix.outputs.matrix) }}
+        kind: [os, enterprise]
+        os: [ ubuntu-latest ]
+        java: [ { version: 17, distribution: corretto },
```

Comment:
Platform tests against the following and we should at least use the ones in this list, both 17 and 21 versions of each:

1. Oracle JDK
2. Zulu JDK
3. Openj9
4. Corretto JDK
5. RedHat JDK
6. Microsoft JDK
7. Zing JDK

Missing Openj9, RedhatJDK, and Zing JDK from the list.

Reply:
We are using the `setup-java` action for this. According to the documentation it supports the following list of distributions: [https://github.com/actions/setup-java?tab=readme-ov-file#supported-distributions](https://github.com/actions/setup-java?tab=readme-ov-file#supported-distributions). I can find Openj9 there, but where can I find RedhatJDK and Zing JDK?

Reply:
Let's at least use the ones available, Oracle, zulu, adopt-openj9, corretto, microsoft. You may need to manually install the RedHat and Zing JDKs. You can use the available ones and open an issue to add the missing ones for the sake of completing this PR or just install and use them as well.